feat(seer): rpc endpoint to get sentry org ID by repo

giovanni-guidini opened the pull request:
adds a new RPC endpoint to get the Sentry org_id from the repo details (full name + provider). This is so that we can use the bug prediction with Sentry Issues lookup when the request comes through GitHub webhooks.

In these cases Overwatch doesn't have (at this time, at least) a way to lookup the sentry org ID. More details [here](https://linear.app/getsentry/issue/PREVENT-224/config-lookup-sentry-organization-id-by-github-orgrepo-slug-using)

<!-- Describe your PR here. -->

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

giovanni-guidini committed:
feat(seer): rpc endpoint to get sentry org ID by repo

adds a new RPC endpoint to get the Sentry org_id from the repo details (full name + provider). This is so that we can use the bug prediction with Sentry Issues lookup when the request comes through GitHub webhooks.

In these cases Overwatch doesn't have (at this time, at least) a way to lookup the sentry org ID.
More details [here](https://linear.app/getsentry/issue/PREVENT-224/config-lookup-sentry-organization-id-by-github-orgrepo-slug-using)

note: we're using `.filter().first()` to avoid problems with multiple repos with the same name,
cause it seems the unique clause in the DB doesn't cover that case.
However I don't think github (and other providers) allows for 2 repos with the same full name?

In any case thanks @rohitvinnakota-codecov and bug-predictor for flagging that

rohitvinnakota-codecov commented on src/sentry/seer/endpoints/seer_rpc.py:
```diff
@@ -213,6 +214,25 @@ def get_organization_slug(*, org_id: int) -> dict:
+        provider: The provider of the repository (e.g. "integrations:github")
+    """
+    try:
+        repo = Repository.objects.get(
+            name=full_repo_name, provider=provider, status=ObjectStatus.ACTIVE
+        )
```
I think this is correct

giovanni-guidini committed:
fix(seer): make the correct query

After some investigation (thanks team for double checking stuff together)
there are some important remarks:
* The exact same repo can be part of multiple organizations
* adding the repo "external_id" to the lookup reduces a lot the number
  of repeated orgs returned (looking at prod)

So the query now requires the repo external ID, and will return all org IDs
- that gave us consent to use AI features - in the response

rohitvinnakota-codecov commented on src/sentry/seer/endpoints/seer_rpc.py:
```diff
@@ -213,6 +214,39 @@ def get_organization_slug(*, org_id: int) -> dict:
+    orgs_with_consent = []
+    for org in organizations:
+        hide_ai_features = org.get_option("sentry:hide_ai_features", HIDE_AI_FEATURES_DEFAULT)
+        pr_review_test_generation_enabled = bool(
```
nit: could be nice to move checks these to helper functions 
`can_use_prevent_ai_features`.

rohitvinnakota-codecov commented on src/sentry/seer/endpoints/seer_rpc.py:
```diff
@@ -213,6 +214,39 @@ def get_organization_slug(*, org_id: int) -> dict:
     return {"slug": org.slug}
 
 
+def get_sentry_organization_id(
```
We should change the fn name

giovanni-guidini committed:
address review comments

* move some logic to a helper function
* change the function name to reflext that we return multiple org IDs now
